File: app/src/main/java/com/sleticalboy/dailywork/util/UIUtils.java
package com.sleticalboy.dailywork.util;

import android.content.Context;
import android.util.DisplayMetrics;
import android.view.WindowManager;

/**
 * Created on 18-2-3.
 *
 * @author sleticalboy
 * @version 1.0
 * @description utils for operations about UI
 */
public class UIUtils {

    /**
     * 获取屏幕的尺寸
     *
     * @param context
     * @return
     */
    public static Size getScreenSize(Context context) {
        WindowManager wm = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
        DisplayMetrics displayMetrics = new DisplayMetrics();
        wm.getDefaultDisplay().getMetrics(displayMetrics);
        float density = displayMetrics.density;
        int width = (int) (displayMetrics.widthPixels * density);
        int height = (int) (displayMetrics.heightPixels * density);
        return new Size(width, height);
    }

    /**
     * dp to px
     *
     * @param context
     * @param dp
     * @return
     */
    public static int dp2px(Context context, float dp) {
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dp * scale + 0.5f);
    }

    /**
     * px to dp
     *
     * @param context
     * @param px
     * @return
     */
    public static int px2dp(Context context, float px) {
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (px / scale + 0.5f);
    }

    /**
     * sp to px
     *
     * @param context
     * @param sp
     * @return
     */
    public static int sp2px(Context context, float sp) {
        final float fontScale = context.getResources().getDisplayMetrics().scaledDensity;
        return (int) (sp * fontScale + 0.5f);
    }

    public static class Size {

        private final int mWidth;
        private final int mHeight;

        Size(int width, int height) {
            mWidth = width;
            mHeight = height;
        }

        public int getWidth() {
            return mWidth;
        }

        public int getHeight() {
            return mHeight;
        }
    }
}

File: MainApp/src/main/java/com/sleticalboy/dailywork/util/UIUtils.kt
package com.sleticalboy.dailywork.util

import android.content.Context
import android.util.DisplayMetrics
import android.view.WindowManager

/**
 * Created on 18-2-3.
 *
 * @author leebin
 * @version 1.0
 * @description utils for operations about UI
 */
object UIUtils {

    /**
     * 获取屏幕的尺寸
     *
     * @param context
     * @return
     */
    fun getScreenSize(context: Context): Size {
        val wm = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        val displayMetrics = DisplayMetrics()
        wm.defaultDisplay.getMetrics(displayMetrics)
        val density = displayMetrics.density
        val width = (displayMetrics.widthPixels * density).toInt()
        val height = (displayMetrics.heightPixels * density).toInt()
        return Size(width, height)
    }

    /**
     * dp to px
     *
     * @param context
     * @param dp
     * @return
     */
    fun dp2px(context: Context, dp: Float): Int {
        val scale = context.resources.displayMetrics.density
        return (dp * scale + 0.5f).toInt()
    }

    /**
     * px to dp
     *
     * @param context
     * @param px
     * @return
     */
    fun px2dp(context: Context, px: Float): Int {
        val scale = context.resources.displayMetrics.density
        return (px / scale + 0.5f).toInt()
    }

    /**
     * sp to px
     *
     * @param context
     * @param sp
     * @return
     */
    fun sp2px(context: Context, sp: Float): Int {
        val fontScale = context.resources.displayMetrics.scaledDensity
        return (sp * fontScale + 0.5f).toInt()
    }

    class Size internal constructor(val width: Int, val height: Int)
}
